.github/workflows/test.yml

```diff
@@ -0,0 +1,31 @@
+name: Test
+
+on: 
```

Comment:
the way i've configured it, the Test and Build workflows are both run on the relevant pull request events, with Build workflow also being run on each merge/push to master (so we can use those artifacts in the release)

if you like this, I'll do the same on the server as well.

Reply:
Actually, on further discussion w/ @bemasc, I'm fleshing something else out: https://docs.google.com/document/d/1MXHfL2pnIlaGf2FGEToOyvJdlPLhDgZjoxOzU3vGCxg/edit?resourcekey=0-mzsmPr03Ez8PtDaipwGrLQ

Will mark this as draft for the moment.